<launch>

    <!-- visualize apollo -->
    <include file="$(find nuturtle_description)/launch/load.launch" >
        <arg name="use_sim" value="true"/>
        <arg name="use_rviz" value="false"/>
    </include>

    <!-- launch nodes -->
    <node name="teleop" pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" />
    <node name="odometer" pkg="rigid2d" type="rigid2d_odometer" />
    <node name="tube_world" pkg="nurtlesim" type="nurtlesim_tube_world" />

    <!-- parameters for nodes -->
    <rosparam param="wheel_base">0.16</rosparam>
    <rosparam param="wheel_radius">0.033</rosparam>
    <rosparam param="left_wheel_joint">wheel_left_joint</rosparam>
    <rosparam param="right_wheel_joint">wheel_right_joint</rosparam>
    <rosparam param="odom_frame_id">odom</rosparam>
    <rosparam param="body_frame_id">base_footprint</rosparam>
    <rosparam param="x_variance">0.01</rosparam>
    <rosparam param="w_variance">0.01</rosparam>
    <rosparam param="slip_min">0</rosparam>
    <rosparam param="slip_max">2</rosparam>
    
    <rosparam file="$(find nurtlesim)/config/simulation_params.yaml"/>

</launch>